hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+
+                    if new_f_ext in  [".shp", ".shx", ".dbf"]:
+                        validate_files_dict['are_files_valid'] = False
+                        validate_files_dict['message'] = "No more shp, shx, dbf files can be added."
+                        some_new_files_added=False
+                        break
+                    elif (new_f_name!=ori_file_info.baseFilename) and \
+                            (not (new_f_name == ori_file_info.baseFilename+".shp" and new_f_ext==".xml")):
+                        #need to check if is ShapefileBaseName.shp.xml
+                        validate_files_dict['are_files_valid'] = False
+                        validate_files_dict['message'] = "At least one file does not follow the ESRI Shapefile naming convention."
+                        some_new_files_added=False
+                        break
+                    elif crt_f_str.find(new_f_fullname)!=-1:
+                        validate_files_dict['are_files_valid'] = False
+                        validate_files_dict['message'] = "At least one file already exists."
+                        some_new_files_added=False
+                        break
+                if some_new_files_added:
+                    ori_fn_dict=json.loads(ori_file_info.filenameString)
+                    for f in files:
+                        new_f_fullname=f.name.lower()
+                        ori_fn_dict[new_f_fullname]="new"
+                    res_obj.metadata.update_element('OriginalFileInfo', element_id=ori_file_info.id,filenameString=json.dumps(ori_fn_dict))
+            else: # all files has been removed, start it over
+                if files:
+                    rslt=check_uploaded_files_type(files)
+
+                    tmp_dir = None
+                    uploaded_file_type = None
+                    baseFilename=None
+                    uploadedFileCount=0
+                    uploadedFilenameString=None
+                    shp_full_path=None
+                    validate_files_dict['are_files_valid'] = rslt['are_files_valid']
+                    validate_files_dict['message'] = rslt['message']
+
+                    if validate_files_dict['are_files_valid']:
+
+                        if res_obj.metadata.originalfileinfo.all().first():
+                            res_obj.metadata.originalfileinfo.all().delete()
+                        if res_obj.metadata.geometryinformation.all().first():
+                            res_obj.metadata.geometryinformation.all().delete()
+                        if res_obj.metadata.fieldinformation.all().first():
+                            res_obj.metadata.fieldinformation.all().delete()
+                        if res_obj.metadata.originalcoverage.all().first():
+                            res_obj.metadata.originalcoverage.all().delete()
+                        if res_obj.metadata.coverages.all().first():
+                            res_obj.metadata.coverages.all().delete()
```

Comment:
No need to query for the object just to delete it, you're only doubling the number of SQL queries you have to use.